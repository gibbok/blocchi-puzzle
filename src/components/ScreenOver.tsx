import * as React from 'react';
import styled from 'styled-components';
import { Button } from './Button';
import { gameSlice } from '../store';
import { useDispatch } from 'react-redux';
import { ScoreContainer } from '../containers';

const {
  actions: { screenGame, resetGame },
} = gameSlice;

const StyledScreenOver = styled.div`
  margin-top: -100%;
`;

const ButtonWrapper = styled.div`
  margin-top: 4rem;
`;

export const ScreenOver = (): JSX.Element => {
  const dispatch = useDispatch();

  const handleClickPlay = () => {
    dispatch(resetGame());
    dispatch(screenGame());
  };

  return (
    <StyledScreenOver>
      <ScoreContainer />
      <ButtonWrapper>
        <Button onClick={handleClickPlay}>Play again!</Button>
      </ButtonWrapper>
    </StyledScreenOver>
  );
};
