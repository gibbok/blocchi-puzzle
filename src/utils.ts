import { Tetro, Board } from './types';

export const logger = (x: Tetro | Board) => {
  console.log(x.reduce((acc, v) => `${acc} ${v.concat()} \n`, ''));
};
