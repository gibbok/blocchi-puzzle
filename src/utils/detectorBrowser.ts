import { detect } from 'detect-browser';

const getPathRelease = (version: string) => version.split('.')[0];

export const detectorBrowser = (): boolean => {
  const browser = detect();
  if (browser && browser.name && browser.version) {
    const name = browser.name;
    const version = browser.version;
    const path = Number(getPathRelease(version));

    switch (name) {
      case 'chrome':
      case 'firefox':
        return path > 80;
      case 'safari':
        return path > 13;
      case 'opera':
        return path > 62;
      default:
        return false;
    }
  } else {
    return false;
  }
};
