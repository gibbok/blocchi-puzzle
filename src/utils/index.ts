export * from './data.support.test';
export * from './logger';
