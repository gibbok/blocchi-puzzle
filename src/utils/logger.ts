import { Tetro, Board } from '../game/types';

export const logger = (x: Tetro | Board): void => {
  console.log(x.reduce((acc, v) => `${acc} ${v.concat()} \n`, ''));
};
