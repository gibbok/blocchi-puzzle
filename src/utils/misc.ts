import { IO, io } from 'fp-ts/lib/IO';
import { randomInt } from 'fp-ts/lib/Random';

export const getRandomValueFromStringEnum = <T>(strEnum: T): IO<T[keyof T]> => {
  const rndInt = randomInt(0, Object.keys(strEnum).length - 1)();
  const rndValue = Object.keys(strEnum)[rndInt] as keyof T;
  return io.of(strEnum[rndValue]);
};
