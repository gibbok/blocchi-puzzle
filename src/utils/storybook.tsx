import React from 'react';
import GlobalStyle from '../components/GlobalStyle';
import styled from 'styled-components';
import { wood } from '../assets/images';

/**
 * Decorate a story with glocal styles.
 * @param story Story
 */
export const globalStylesDecorator = (story: () => unknown): JSX.Element => (
  <React.Fragment>
    <GlobalStyle />
    {story()}
  </React.Fragment>
);

const Bkg = styled.div`
  background-image: url(${wood});
  width: 100vw;
  height: 100vh;
`;

/**
 * Decorate a story with a page background.
 * @param story Story
 */
export const pageBackground = (story: () => React.ReactNode): JSX.Element => <Bkg>{story()}</Bkg>;
