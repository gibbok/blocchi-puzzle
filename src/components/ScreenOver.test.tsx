import { ScreenOver } from './ScreenOver';
import renderer from 'react-test-renderer';
import React from 'react';
import { mockStore } from '../utils';
import { Provider } from 'react-redux';

describe('ScreenOver', () => {
  it('should render', () => {
    const store = mockStore();
    const tree = renderer
      .create(
        <Provider store={store}>
          <ScreenOver />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
