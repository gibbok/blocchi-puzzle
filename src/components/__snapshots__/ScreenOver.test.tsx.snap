// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenOver should render 1`] = `
<div
  className="sc-fzqNJr dCjklm"
>
  <div
    className="sc-fzqBZW ckAzKi"
  >
    <div
      className="sc-fznKkj dBOSXt"
    >
      <div
        className="sc-fznZeY ihsglE"
      >
        Score
      </div>
      <div
        className="sc-fznZeY ihsglE"
      >
        0
      </div>
    </div>
    <div
      className="sc-fznKkj dBOSXt"
    >
      <div
        className="sc-fznZeY ihsglE"
      >
        Level
      </div>
      <div
        className="sc-fznZeY ihsglE"
      >
        1
      </div>
    </div>
  </div>
  <div
    className="sc-fzoyAV fqmqGy"
  >
    <button
      className="sc-AxjAm dXSKqi"
      onClick={[Function]}
    >
      <div
        className="sc-AxirZ kCiwbh"
      >
        Play again!
      </div>
    </button>
  </div>
</div>
`;
